contributing.md

```diff
@@ -0,0 +1,163 @@
+
+1. Most changes to the code start when a user makes an issue in the pylabrad issue tracker.
+This issue explains what feature is to be added or what bug is to be fixed.
+The issue is automatically assigned a unique number
+
+1. Development proceeds in a branch
+ * The branch name is `XX-description-of-issue` where `XX` is the issue number and everything else is a brief description of the issue.
+For example, the issue filed in which we stated the need for this contributing document was #134, so the branch was called `134-contributing`.
+ * If the code change is particularly urgent (for example, a major bug was introduced in the previous commit), you can add the `hotfix` label to the pull request (or the issue, or both).
+ * Reviewers are strongly urged to review `hotfix` labeled items at the earliest possible opportunity.
+
+1. Development is done in the branch via a series of commits.
+The commits do not have to be done all by the same person, but if multiple people are working on an issue in the same branch common sense says they should communicate well to ensure smooth workflow.
+
+1. When the feature is nearly complete or the bug is fixed, a pull request is filed, usually by whoever wrote the new code.
+This notifies other users that some new code is ready for review.
+The user making the pull request can and should call out specific users who are qualified to review the code.
+Those users review the code by reading the changes and running the new tests.
+Comments on specific commits can be made via the github pull request page.
+Try to focus comments on specific commits, but it's ok to put general discussion in the main pull request comment area.
+
+1. Any user may add more commits to address outstanding issues with the code, but normally this should be done by the person who made the pull request.
+
+1. Once you think the code is in good shape, comment "LGTM" (looks good to me) in the general comments area on the pull request.
+This indicates that it is ok to merge the feature branch into master.
+Writing LGTM on a pull requests indicates that you are convinced the code works as expected and that you now share responsibility with the author for any problems arising from the change.
+**Only the person who filed the pull request may actually do the merge**.
+For larger or more critical changes it is advised to get several LGTM's before merging.
+
+1. Tests should be written for any new functionality. Here are a few guidelines for writing tests:
+  * Tests for a package should exist in a subpackage of the package. For example, tests for modules in
+    labrad.servers should go in labrad/servers/test.
+  * Tests should stand alone as much as possible.
+  * When in doubt, testing something is better than testing nothing.
+
+  Finally, any changes to pylabrad should pass all existing tests. You can check this by running ```py.test``` from the command line when in the pyle directory.
```

Comment:
Should add a note here that the tests currently require a labrad manager to be running along with a python test server. In the future this can be improved (e.g. separate unit and integration tests, with only integration tests requiring other services).